index.js

```diff
@@ -375,6 +379,29 @@ class GithubScm extends Scm {
     }
 
     /**
+     * Check validity of github webhook event signature
+     * @method _checkSignature
+     * @param   {String}    secret      The secret used to sign the payload
+     * @param   {String}    payload     The payload of the webhook event
+     * @param   {String}    signature   The signature of the webhook event
+     * @returns {boolean}
+     */
+    _checkSignature(secret, payload, signature) {
+        const hmac = crypto.createHmac('sha1', secret);
+
+        hmac.setEncoding('hex');
+        hmac.write(JSON.stringify(payload), 'utf-8');
```

Comment:
Interesting, I wonder if this is the same as hmac.write(new Buffer(text, 'utf-8'));